### PR TITLE
BUG: fix kendalltau issue where p-value becomes >1

### PR DESCRIPTION
**Pull request by Konrad0**
Fixes the issue of stats.kendalltau(), where in some cases a
p-value larger than 1 was returned.

#### Reference issue
Fixes issue #11175 

#### What does this implement/fix?
Check added for the special case of concordance-discordance around zero.

### CODE REVIEW

File: scipy/stats/mstats_basic.py [hunk 1]
```diff
@@ -622,6 +622,8 @@ def kendalltau(x, y, use_ties=True, use_missing=False, method='auto'):
             prob = 2.0/np.math.factorial(n)
         elif c == 1:
             prob = 2.0/np.math.factorial(n-1)
+        elif 2*c == (n*(n-1))/2:
```
Comment (rgommers) [inline scipy/stats/mstats_basic.py]: Looks like this should be integer division (`//`), could you change that?

**Review by rgommers — COMMENTED** [review]
Thanks @Konrad0. LGTM module my one minor comment.

File: scipy/stats/tests/test_stats.py [hunk 2]
```diff
@@ -880,6 +880,47 @@ def test_tie2(self):
 
 
 def test_kendalltau():
+
+    # case without ties, con-dis equal zero
+    x = [5,2,1,3,6,4,7,8]
```
Comment (chrisb83) [inline scipy/stats/tests/test_stats.py]: Please add a space after a comma in your code (PEP8)

**Review by chrisb83 — COMMENTED** [review]
Just a minor PEP8 comment. Test cases against R look good. 

So this is just a special case that needs separate treatment and the rest of the code does not require any changes?

**Comment by Konrad0** [comment]
Hi @rgommers and @chrisb83,

thanks for your comments. Yes, to my understanding, this issue does not require any further treatment in the rest of the code. Your improvements are both implemented in my latest commit.

BTW, in case you're interested, I checked the R documentation and their calculation of the p-value in absence of ties seems to be based on "Algorithm AS 71: The Upper Tail Probabilities of Kendall's Tau", D. J. Best and P. G. Gipps, Journal of the Royal Statistical Society. Series C (Applied Statistics), Vol. 23, No. 1 (1974), pp. 98-100, DOI: 10.2307/2347062, https://www.jstor.org/stable/2347062. This publication includes an Edgeworth series improving the asymptotic normal approximation we might want to consider. The exact method looks similar to ours, but the R code is recursive.

**Comment by rgommers** [comment]
Merged, thanks @Konrad0!

**Comment by mdhaber** [comment]
Thanks @Konrad0! If you'd like your name added to [THANKS.txt](https://github.com/scipy/scipy/blob/master/THANKS.txt), please make an entry [here](https://github.com/scipy/scipy/wiki/*THANKS.txt-additions-modifications-for-1.5.0*).

File: scipy/stats/mstats_basic.py [hunk 3]
```diff
@@ -622,6 +622,8 @@ def kendalltau(x, y, use_ties=True, use_missing=False, method='auto'):
             prob = 2.0/np.math.factorial(n)
         elif c == 1:
             prob = 2.0/np.math.factorial(n-1)
+        elif 2*c == (n*(n-1))//2:
+            pvalue = 1.0
```
Comment (WarrenWeckesser) [inline scipy/stats/mstats_basic.py]: This should be `prob = 1.0`.  Test case:

```
from scipy.stats import mstats

x = [9, 2, 5, 6]
y = [4, 7, 9, 11]
result = mstats.kendalltau(x, y, method='exact')
```
With scipy master branch (version '1.5.0.dev0+fac296a'), we currently get
```
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-8-0ed9ea83fd63> in <module>
----> 1 mstats.kendalltau(x, y, method='exact')

~/mc37scipy/lib/python3.7/site-packages/scipy-1.5.0.dev0+fac296a-py3.7-macosx-10.7-x86_64.egg/scipy/stats/mstats_basic.py in kendalltau(x, y, use_ties, use_missing, method)
    666         raise ValueError("Unknown method "+str(method)+" specified, please use auto, exact or asymptotic.")
    667 
--> 668     return KendalltauResult(tau, prob)
    669 
    670 
UnboundLocalError: local variable 'prob' referenced before assignment
```

**Comment by WarrenWeckesser** [comment]
@Konrad0, I just noticed a bug introduced by this PR.  Do you think you could create another PR to fix it?

**Comment by Konrad0** [comment]
Thanks for catching the bug @WarrenWeckesser, sorry for introducing it and being unavailable to fix it for so long. I created PR  #11489 which should fix the problem and added your example as a test (checked against R).

**Pull request merged**